lib/msal-native-auth/src/auth_flow/result/ResetPasswordResult.ts

```diff
@@ -0,0 +1,112 @@
+/*
```

Comment:
This class I think should be refactor to a single state handler for reset password:
```
class ResetPasswordResult
```
This might be longer but we can discuss online.

Having a single class that holds the overall state makes it easier to extend and change.

Reply:
Each reset password result is not created based on its state. Those result classes are created and named by the pattern: `{Action}Result`. For example:
- ResetPasswordResendCodeResult: this is the result for the resend code action.
- ResetPasswordStartResult: this is the result for the reset password start action.
- ResetPasswordSubmitPasswordResult: this is the result for the submit password action.

Also for each result, it may have the different states. The logic of this implementation is used by all sign-in, sign-up, password-reset actions. So I will suggest using this for now if you don't have strong opinion since the existing approach may leave more space for the further new features (for example, any result may have a different result data type).

Reply:
But you see the above, should not be classes, but just specific interfaces for given responses. We know from the API documentation that the API calls for each of the above returns a response that has a specific format, and the SDK should map them 1-1, ex I would do these interfaces. Example the interfaces I am creating for Sign Up map the Results (Response) to the API:

```

/**
 * Interface for redirect challenge response
 */
export interface RedirectChallengeResponse {
    challenge_type: ChallengeType.REDIRECT;
}

export interface SignUpChallengeResponse {
    interval: number;
    continuation_token: string;
    challenge_type: ChallengeType.OOB;
    binding_method: BindingMethod;
    challenge_channel: ChallengeChannel;
    challenge_target_label: string;
    code_length: number;
}

export interface SignUpStartResponse {
    continuation_token?: string;
}

/**
 * Union type for all possible challenge responses
 */
export type ChallengeResponse =
    | SignUpChallengeResponse
    | RedirectChallengeResponse;

export interface SignUpContinueResponse {
    challenge_type: ChallengeType;
    continuation_token: string;
}
```

Then all these responses can be mapped with by single `SignUpResult.ts` class as wrapper, which will return the correct state result based on the response type.

Reply:
Please note these result classes are not for the API response. All the API response classes are placed in the network_client/response folder. Those result classes are used for the actions of SDK.